src/initialization.jl

```diff
@@ -258,50 +258,58 @@ function get_initial_values(prob, valp, f, alg::OverrideInit,
             initdata.update_initializeprob!(initprob, valp)
         end
     end
+    nlsol, success = solve_initialization(initdata, initprob, alg; reltol, abstol, nlsolve_alg )
 
-    if is_trivial_initialization(initdata)
-        nlsol = initdata
-        success = true
-    else
-        nlsolve_alg = something(nlsolve_alg, alg.nlsolve, Some(nothing))
-        if nlsolve_alg === nothing && state_values(initprob) !== nothing
-            throw(OverrideInitMissingAlgorithm())
-        end
-        if alg.abstol !== nothing
-            _abstol = alg.abstol
-        elseif abstol !== nothing
-            _abstol = abstol
-        else
-            throw(OverrideInitNoTolerance(:abstol))
-        end
-        if alg.reltol !== nothing
-            _reltol = alg.reltol
-        elseif reltol !== nothing
-            _reltol = reltol
-        else
-            throw(OverrideInitNoTolerance(:reltol))
-        end
-        nlsol = solve(initprob, nlsolve_alg; abstol = _abstol, reltol = _reltol, kwargs...)
-
-        success = if initprob isa NonlinearLeastSquaresProblem
-            # Do not accept StalledSuccess as a solution
-            # A good local minima is not a success 
-            resid = nlsol.resid
-            normresid = norm(resid)
-            SciMLBase.successful_retcode(nlsol) && normresid <= abstol
-        else
-            SciMLBase.successful_retcode(nlsol)
-        end
-    end
-
+    nlsol2 = prob.f.initialization_data.initializeprob
     if initdata.initializeprobmap !== nothing
-        u0 = initdata.initializeprobmap(choose_branch(nlsol))
+        u02 = initdata.initializeprobmap(nlsol2)
```

Comment:
This doesn't make sense - we solve initialization, but calculate the new `u0` from the unsolved problem?